core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -389,11 +389,16 @@ protected Node visitShowColumns(ShowColumns showColumns, Void context)
             if (!metadata.schemaExists(session, new CatalogSchemaName(tableName.getCatalogName(), tableName.getSchemaName()))) {
                 throw semanticException(SCHEMA_NOT_FOUND, showColumns, "Schema '%s' does not exist", tableName.getSchemaName());
             }
-            if (metadata.getView(session, tableName).isEmpty() &&
-                    metadata.getTableHandle(session, tableName).isEmpty()) {
+            Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+            if (metadata.getView(session, tableName).isEmpty() && tableHandle.isEmpty()) {
                 throw semanticException(TABLE_NOT_FOUND, showColumns, "Table '%s' does not exist", tableName);
             }
 
+            if (tableHandle.isPresent()) {
+                // We are using information_schema which could ignore errors when getting a list of columns.
+                // In order to not to swallow any issue getTableMetadata() is called.
+                metadata.getTableMetadata(session, tableHandle.get());
```

Comment:
Document why you still go thru information schema despite having all information at hand

Reply:
@kokosing do we rely here on 
```
    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
...
            if (columns.isEmpty()) {
                // A table may have no supported columns. In rare cases (e.g. PostgreSQL) a table might have no columns at all.
                throw new TableNotFoundException(
                        schemaTableName,
                        format("Table '%s' has no supported columns (all %s columns are not supported)", schemaTableName, allColumns));
            }
```
https://github.com/trinodb/trino/blob/master/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java#L313-L318
It's far from obvious. Worth to underline it somehow in comment.

Reply:
@ssheikin Not only. What you show here is just one example. The issue is more generic than JDBC connectors only and affects other connectors.